ui/src/app/utils/access-utils.tsx

```diff
@@ -361,6 +361,52 @@ export const useIsUserDisabled = () => {
   return disabled;
 };
 
+export const useNeedsToAcceptTOS = () => {
+  const { authLoaded, isSignedIn } = useStore(authStore);
+  const [userRequiredToAcceptTOS, setUserRequiredToAcceptTOS] = useState<
+    boolean | undefined
+  >(false);
+  useEffect(() => {
+    if (!authLoaded) {
+      return;
+    }
+
+    let mounted = true;
+    if (!isSignedIn) {
+      setUserRequiredToAcceptTOS(false);
+    } else {
+      (async () => {
+        try {
+          await profileStore.get().load();
+          if (mounted) {
```

Comment:
why do we have different behavior when it's mounted?

Reply:
removed it, i used it because it was being used in hook for userDisable, But i could not reproduce the scenarios where its different